sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java

```diff
@@ -353,8 +353,9 @@ private void initDataReader(
       throw new IOException("could not read page in col " + descriptor, e);
     }
     // for PARQUET-246 (See VectorizedDeltaByteArrayReader.setPreviousValues)
-    if (CorruptDeltaByteArrays.requiresSequentialReads(writerVersion, dataEncoding) &&
-        previousReader instanceof RequiresPreviousReader) {
+    if (CorruptDeltaByteArrays.requiresSequentialReads(writerVersion, dataEncoding)
+        && previousReader != null
+        && dataColumn instanceof RequiresPreviousReader) {
```

Comment:
Could you file a JIRA issue, @yutsareva ?

Reply:
done https://issues.apache.org/jira/browse/SPARK-48234